pkg/coscheduling/core/core.go

```diff
@@ -112,7 +119,37 @@ func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, pod *corev1.Pod) er
 		return fmt.Errorf("cannot found engough pods, "+
 			"current pods number: %v, minMember of group: %v", len(pods), pg.Spec.MinMember)
 	}
-	return nil
+
+	// TODO(cwdsuzhou): This resource check can not absolutely to avoid unschedulable group. We should add Post-Filter to reject all if anyone does not fit.
+	nodes, err := pgMgr.snapshotSharedLister.NodeInfos().List()
+	if err != nil {
+		return err
+	}
+	if !pgMgr.enableResourceCheck {
+		return nil
+	}
+	if _, ok := pgMgr.permittedPG.Get(pgFullName); ok {
+		return nil
+	}
+	matchCount := 0
+	for _, node := range nodes {
+		nodeCopy := node.Clone()
+		for {
```

Comment:
Shouldn't we check if matchCount is not less than minAvailable here? Otherwise, the for loop above may have been run way more than it ought to, right?

```go
for matchCount < int(pg.Spec.MinMember) {
    ...
}
```

Reply:
Good catch, thanks.